DigitalLearningSolutions.Data.Tests/Services/TrackerActionServiceTests.cs

```diff
@@ -136,5 +142,63 @@ public void GetObjectiveArrayCc_returns_empty_object_json_if_no_results_found()
                 )
             ).MustNotHaveHappened();
         }
+
+        [Test]
+        public void StoreDiagnosticJson_returns_success_response_if_successful()
+        {
+            // Given
+            const int progressId = 1;
+            const string diagnosticOutcome = "[{'tutorialid':425,'myscore':4},{'tutorialid':424,'myscore':3}]";
+
+            A.CallTo(
+                () => progressService.UpdateDiagnosticScore(
+                    A<int>._,
+                    A<int>._,
+                    A<int>._
+                )
+            ).DoesNothing();
+
+            // When
+            var result = trackerActionService.StoreDiagnosticJson(progressId, diagnosticOutcome);
+
+            // Then
+            result.Should().Be(TrackerEndpointResponse.Success);
+            A.CallTo(
+                () => progressService.UpdateDiagnosticScore(
+                    1,
+                    424,
+                    3
+                )
+            ).MustHaveHappenedOnceExactly();
+            A.CallTo(
+                () => progressService.UpdateDiagnosticScore(
+                    1,
+                    425,
+                    4
+                )
+            ).MustHaveHappenedOnceExactly();
+        }
+
+        [Test]
+        [TestCase(null, "[{'tutorialid':425,'myscore':4},{'tutorialid':424,'myscore':3}]")]
+        [TestCase(1, null)]
+        public void StoreDiagnosticJson_returns_StoreDiagnosticScoreException_if_parameter_missing(

```

Comment:
Can we also have a test for the exception case where it goes through the catch block with some error?

Reply:
added one!